File: java/src/beginner/NumberOfSteps.java
package beginner;

public class NumberOfSteps {
  public static int numberOfSteps(int num) {
    int count = 0;
    int val = num;
    while (val > 0) {
      count += 1;
      val = val % 2 == 0 ? val / 2 : val - 1;
    }
    return count;
  }

  public static void main(String[] args) {
    System.out.println(numberOfSteps(14));
  }
}

File: js/src/beginner/number_of_steps.js
import { strict as assert } from 'node:assert'
import logger from '../utils.js'

function numberOfSteps(num) {
  let count = 0
  while (num > 0) {
    count++
    Number(num & 1) === 0 ? (num >>= 1) : num--
  }
  return count
}

export function test() {
  logger.log('Running test for numberOfSteps')
  assert.deepEqual(numberOfSteps(14), 6)
  assert.deepEqual(numberOfSteps(8), 4)
  assert.deepEqual(numberOfSteps(123), 12)
  logger.log('All test passed!')
}
